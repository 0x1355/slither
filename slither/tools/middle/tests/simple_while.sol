pragma solidity >=0.4.16 <0.7.0;

contract SimpleWhile {
    uint public a;

    function f(uint input) public {
        while (input >= 0) {
            input = input - 1;
            a++;
        }
    }
}