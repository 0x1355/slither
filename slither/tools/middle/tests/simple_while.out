OVERLAY_FUNCTION_
    ENTRY_POINT
    EXPRESSION
    EXPRESSION
    IF_LOOP
    OVERLAY
f
    ENTRY_POINT
    BEGIN_LOOP
    IF_LOOP
    OVERLAY
    END_LOOP
